hybris/common/hooks.c

```diff
@@ -2239,6 +2243,46 @@ static void *_hybris_hook___get_tls_hooks()
     return tls_hooks;
 }
 
+struct __wrapped_atexit {
+    void (*dtor)(void *);
+    void *obj;
+    void *dso_handle;
+};
+
+static void __dtor_wrapper(void *obj) {
+    struct __wrapped_atexit *wrapped = obj;
+    /* Call the wrapped dtor. */
+    wrapped->dtor(wrapped->obj);
+    /* Reduce dso_handle's ref count. */
+    __hybris_remove_thread_local_dtor(wrapped->dso_handle);
+    /* Free the wrapper. */
+    free(wrapped);
+}
+
+extern const void * const __dso_handle;
+
+static int _hybris_hook___cxa_thread_atexit(void (*dtor)(void *), void *obj,
+                                            void *dso_symbol)
+{
+    struct __wrapped_atexit *wrapped = malloc(sizeof(struct __wrapped_atexit));
+    wrapped->dtor = dtor;
+    wrapped->obj = obj;
+    wrapped->dso_handle = dso_symbol;
+
+    /* Call Glibc's implementation. Pass our symbol to prevent ourselve from
```

Comment:
Nitpick: typo in ourselve, probably should be ourself.

Reply:
@peat-psuwit can you fix the typo to satisfy @mlehtima , then i think we can merge it

Reply:
Sorry, forgot about that. Will fix it soon.

Reply:
Fixed.